test_bst.py

```diff
@@ -0,0 +1,91 @@
+import bst
+import pytest
+
+
+def test_Bst():
+    """Constructing a bst tree"""
+    tree = bst.Bst()
+    assert tree.tree == {}
+    tree = bst.Bst(5)
+    assert tree.tree == {5: {'depth': 1}}
+
+
+def test_insert(empty_tree):
+    """Test insert into a tree"""
+    tree = empty_tree
+    tree.insert(20)
+    assert tree.tree == {20: {'depth': 1}}
+    tree.insert(30)
+    expect = {20: {'depth': 1,
+                   'right': 30},
+              30: {'depth': 2}}
+    assert tree.tree == expect
+    tree.insert(40) 
+    expect = {20: {'depth': 1,
+                   'right': 30},
+              30: {'depth': 2,
+                   'right': 40},
+              40: {'depth': 3}}
+    assert tree.tree == expect
+    tree.insert(10)
+    expect = {20: {'depth': 1,
+                   'left': 10,
+                   'right': 30},
+              30: {'depth': 2,
+                   'right': 40},
+              40: {'depth': 3},
+              10: {'depth': 2}}
+    assert tree.tree == expect
```

Comment:
This test is too long.  make some support methods that do the work, then call them.  Or break this test into a series of tests: `test_insert_root`, `test_insert_right_child`, test_insert_left_child`, `test_insert_left_grandchild`, etc.